internal/engine/upper.go

```diff
@@ -222,6 +225,19 @@ func handleBuildStarted(ctx context.Context, state *store.EngineState, action Bu
 	removeFromTriggerQueue(state, mn)
 }
 
+func handleResetRestarts(ctx context.Context, engineState *store.EngineState, a server.ResetRestartsAction) {
```

Comment:
it seems very weird for this action to be in the `server` package, given that it has nothing to do with servers

Reply:
It can't go with the other actions in engine because then there's an import cycle:

```
import cycle not allowed
package github.com/windmilleng/tilt/cmd/tilt
	imports github.com/windmilleng/tilt/internal/cli
	imports github.com/windmilleng/tilt/internal/demo
	imports github.com/windmilleng/tilt/internal/engine
	imports github.com/windmilleng/tilt/internal/hud/server
	imports github.com/windmilleng/tilt/internal/engine
```

Reply:
Open to other suggestions for where to put it though.

Reply:
can you put it in store?

would also be open to a store/action package or smth

Reply:
ah I didn't realize we had already had actions in different packages. Putting it in store worked.